source/h2_connection.c

```diff
@@ -270,14 +279,14 @@ static struct aws_h2_connection *s_connection_new(
         goto error;
     }
 
-    if (aws_hash_table_init(
-            &connection->thread_data.closed_streams_where_frames_might_trickle_in,
+    if (aws_lru_cache_init(
+            &connection->thread_data.closed_streams,
             alloc,
-            8,
             aws_hash_ptr,
             aws_ptr_eq,
             NULL,
-            NULL)) {
+            s_stream_closed_detail_destroy,
+            connection->thread_data.max_closed_stream_cache_size)) {
```

Comment:
why not just pass AWS_H2_DEFAULT_MAX_CACHE_SIZE directly and get N * 8 bytes back for our customer.

Reply:
Yeah, you are right.